tasks/Task.cpp

```diff
@@ -72,6 +77,10 @@ void Task::updateHook()
         return;
     }
 
+    /**
+     * Write received state to output port
+     * Only send control command after receiving a valid frame
```

Comment:
This is not precise enough. After readFrame, you have a valid frame. What you want is a valid frame coming from the generator, and only of type PACKET_GENERATOR_STATE_AND_MODEL or PACKET_RUN_TIME_STATE.